frontend/src/components/common/LogViewer.tsx

```diff
@@ -61,12 +63,27 @@ export function LogViewer(props: LogViewerProps) {
   }
 
   React.useEffect(() => {
-    // @todo: Only scroll down automatically if the view hasn't been scrolled up by the user yet.
-    if (logsBottomRef && logsBottomRef.current) {
+    if (logsBottomRef?.current && !isScrolledUp) {
       logsBottomRef.current.scrollIntoView();
     }
   }, [logs]);
 
+  function handleScroll(event: any) {
+    if (logsBottomRef?.current) {
```

Comment:
Reading this, I'm confused what this is supposed to do...

How about adding the description from the issue?
```
- By default the log viewer shows the logs as they happen, scrolling to the bottom automatically like a terminal view
- If the user scrolls the view up, even the tiniest bit, it should remain showing that portion of the log even if more logs arrive
- If the user scrolls down to the bottom again, then new updates will keep the view at the bottom (i.e. it will show the latest contents)
```

Reply:
updated